src/expand/mod.rs

```diff
@@ -0,0 +1,622 @@
+// Copyright (c) 2021  Brendan Molloy <brendan@bbqsrc.net>,
+//                     Ilya Solovyiov <ilya.solovyiov@gmail.com>,
+//                     Kai Ren <tyranron@gmail.com>
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! [Cucumber Expressions][0] [AST] into [`Regex`] transformation
+//! definitions.
```

Comment:
We also need to mention somewhere [upstream production rules](https://github.com/cucumber/cucumber-expressions/blob/main/ARCHITECTURE.md#production-rules).

If we align with them 1-to-1, then they should be referred in transformation functions docs directly.